src/tests/ov_helpers/ov_snippets_models/src/subgraph_mlp.cpp

```diff
@@ -0,0 +1,260 @@
+    auto A_param = std::make_shared<ov::op::v0::Parameter>(precisions[0], input_shapes[0]);
+    std::shared_ptr<Node> A = A_param;
+    if (precisions[0] != ov::element::u8) {
+        A = std::make_shared<ov::op::v0::Convert>(A, ov::element::u8);
+    }
+    auto b_shape = ov::Shape{static_cast<unsigned long>(input_shapes[0][1].get_length()),
+                             static_cast<unsigned long>(input_shapes[0][1].get_length())};
+    auto b_row = ov::Shape{static_cast<unsigned long>(input_shapes[0][1].get_length())};
+
+    // Create subgraph parameters (must be preserved even if similar constants exist).
+    auto sub_A = std::make_shared<ov::op::v0::Parameter>(ov::element::f32, input_shapes[0]);
+    ov::ParameterVector subgraph_params = {sub_A};
+
+    ov::NodeVector subgraph_nodes = {A};
+
+    ov::builder::subgraph::FakeQuantizeOnData onData = {
+        256, {1, 1}, {0.0f}, {2.55f}, {0.f}, {255.f}, ov::element::u8
+    };
+
+    auto decomposed_fq = [](const ov::Output<ov::Node>& input,
+                            const ov::element::Type& out_precision,
+                            float il, float ih, float scale) -> std::shared_ptr<ov::Node> {
+        auto input_low = ov::op::v0::Constant::create(input.get_element_type(), {1, 1}, {il});
+        auto input_high = ov::op::v0::Constant::create(input.get_element_type(), {1, 1}, {ih});
+        auto output_scale = ov::op::v0::Constant::create(input.get_element_type(), {1, 1}, {scale});
+        auto max_node = std::make_shared<ov::op::v1::Maximum>(input, input_low);
+        auto min_node = std::make_shared<ov::op::v1::Minimum>(max_node, input_high);
+        return std::make_shared<ov::op::v1::Multiply>(min_node, output_scale);
+    };
+
+    std::shared_ptr<ov::Node> current = sub_A;
+    current = std::make_shared<ov::snippets::op::ConvertSaturation>(current, ov::element::f32);
+
+    auto mlp_layer = [&](size_t m, size_t n, bool trans) {
+        auto b_shape = ov::Shape{m, n};
+        auto b_row = ov::Shape{m};
+        auto b_shape_trans = ov::Shape{n, m};
+
+        auto dq_shifts = std::make_shared<ov::op::v0::Parameter>(ov::element::f32, b_row);
+        auto dq_shifts_const =
+            std::make_shared<ov::op::v0::Constant>(ov::element::f32, b_row, std::vector<float>{0.1122f});
+        auto dq_scales = std::make_shared<ov::op::v0::Parameter>(ov::element::f32, b_row);
+        auto dq_scales_const =
+            std::make_shared<ov::op::v0::Constant>(ov::element::f32, b_row, std::vector<float>{0.0f});
+
+        auto B = std::make_shared<ov::op::v0::Parameter>(ov::element::i8, b_shape_trans);
+        auto B_const = std::make_shared<ov::op::v0::Constant>(ov::element::i8, b_shape, std::vector<float>{0.0f});
+        auto B_const_trans =
+            std::make_shared<ov::op::v1::Transpose>(B_const,
+                                                    ov::op::v0::Constant::create(ov::element::i32, {2}, {1, 0}));
```

Comment:
Minor: can I ask you to change initialization order to be aligned with topological order? B -> scales -> shifts

Reply:
Reordered